inc/datahandlers/user.php

```diff
@@ -1783,4 +1783,61 @@ function clear_profile($delete_uids=false, $gid=0)
 			remove_avatars($uid);
 		}
 	}
+
+	public function verify_signature()
+	{
+		global $mybb, $parser;
+
+		$parser_options = array(
+			'allow_html' => $mybb->settings['sightml'],
+			'filter_badwords' => 1,
+			'allow_mycode' => $mybb->settings['sigmycode'],
+			'allow_smilies' => $mybb->settings['sigsmilies'],
+			'allow_imgcode' => $mybb->settings['sigimgcode'],
+			"filter_badwords" => 1
+		);
+
+		if($mybb->user['showimages'] != 1 && $mybb->user['uid'] != 0)
```

Comment:
This is incorrect. Datahandler shouldn't assume these values are from the current user. You should pass them as user data.